src/rpc/blockchain.cpp

```diff
@@ -958,14 +959,29 @@ static RPCHelpMan getblock()
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
         }
 
+        if (IsBlockPruned(pblockindex)) {
+            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+        }
+
+        if (verbosity <= 0 && !RPCSerializationFlags()) {
+            if (!ReadRawBlockFromDisk(raw_block, pblockindex, Params().MessageStart())) {
+                throw JSONRPCError(RPC_MISC_ERROR, "Block not found on disk");
+            }
+        }
+
         block = GetBlockChecked(pblockindex);
```

Comment:
Why read the block from disk twice when `verbosity <= 0` ?
1. in the `if (verbosity <= 0 && !RPCSerializationFlags()) {}` to `raw_block`
2. with `block = GetBlockChecked(pblockindex);`

Reply:
Good point. Needs an else here.

Reply:
Fixed